### ISSUE TITLE
Bases of a dynamically created class are not recognized

### ISSUE THREAD

**Issue by m-pilia** [astral-sh/ty#1544]
### Summary

Minimal example:
```python
class Foo:
    pass

x: type[Foo] = type("MyType", (Foo,), {})
```
Result:
```
$  ty check ty_repr.py
error[invalid-assignment]: Object of type `type` is not assignable to `type[Foo]`
 --> ty_repr.py:4:1
  |
2 |     pass
3 |
4 | x: type[Foo] = type("MyType", (Foo,), {})
  | ^
  |
info: rule `invalid-assignment` is enabled by default
```
Expected result: No error should be reported.

See also playground: https://play.ty.dev/5576a11d-e6af-49c8-a08e-fc29816acb44

### Version

ty 0.0.1-alpha.26 (b225fd8b4 2025-11-10)

**Comment by AlexWaygood** [comment]
Thanks! Please see #740
